website/content/docs/concepts/acl/auth-methods/oidc.mdx

```diff
@@ -74,6 +74,107 @@ Complete the login via your OIDC provider. Launching browser to:
+    "Audience": ["https://your-keycloak-instance.com/realms/nomad"],
+    "KeySource": "nomad"
+  }
+}
+```
+
+In this example for Keycloak, Nomad signs the JWT with its own internal private
+key. It sets the JWT's "kid" header as the key ID, as presented by Nomad's
+[jwks.json][] endpoint.
+
+This is arguably the most secure option, because only Nomad has the private key.
+
+Notice the distinction between the two "audience" fields:
+* `BoundAudiences` is often the application client ID (Nomad being the client),
+  which Nomad will verify against what the OIDC provider sends to Nomad. Nomad
+  uses this to make sure that requests are for Nomad, and not some other client.
+  This applies to all OIDC configuration, not only client assertions.
+* `OIDCClientAssertion.Audience` is the OIDC provider, because that is the
+  target audience of the client assertion JWT. The provider uses this to make
+  sure that requests are for it and not some other provider. This is often the
+  same as the `OIDCDiscoveryURL`, so defaults to that.
+
+This option requires that the OIDC provider have network access to Nomad's JWKS,
+either directly or via proxy, but otherwise requires no extra management of key
+material beyond Nomad's built-in [keyring][key-management].
+
+#### User provided key
+
+```json
+{
+  "OIDCDiscoveryURL": "https://login.microsoftonline.com/{tenant}/v2.0",
+  "OIDCClientID": "{app-client-id}",
+  "BoundAudiences": ["{app-client-id}"],
+  "OIDCClientAssertion": {
+    "KeySource": "private_key",
+    "KeyAlgorithm": "RS256",
+    "PrivateKey": {
+      "PemKey": "-----BEGIN RSA PRIVATE KEY-----\nMIIE...the-rest-of-the-key...uJ8fR\n-----END RSA PRIVATE KEY-----",
+      "PemCert": "-----BEGIN CERTIFICATE-----\nMIID...the-rest-of-the-cert...GUCk=\n-----END CERTIFICATE-----"
+    }
+  }
+}
+```
+
+In this example for Microsoft Entra ID (formerly Azure Active Directory),
+we use a private RSA key that we generated separate from Nomad to sign the JWT.
+We also provide an x509 certificate created from the key (or a CA), which Nomad
+uses to derive an [x5t#S256][] thumbprint header. The certificate also must be
+uploaded to the Entra ID app, so that when you try to log in, Entra ID can use
+the "x5t#S256" header to look up the public key that it has stored.
+
+The key and/or certificate may also be configured as filenames on disk on Nomad
+servers with the `PemKeyFile` and `PemCertFile` options, respectively. These
+allow you to rotate your key/cert without needing to update the auth method,
+but they must be present on the disk of any server that may become Nomad leader.
+
+Or, the `KeyID` may be provided directly, instead of a certificate, depending on
+your OIDC provider's requirements.
+
+This allows you to bring your own RSA key for OIDC providers that do not
+support JWKS, network topologies that do not allow connectivity between the
+provider and Nomad JWKS, or if you want a signing key that is specifically
+and only for this purpose.
```

Comment:
```suggestion
Depending on your OIDC provider's requirements, you may provide the `KeyID`
directly, instead of providing a certificate. This approach lets you to bring
your own RSA key for the following scenarios:

- OIDC providers that do not support JWKS
- Network topologies that do not allow connectivity between the provider and
  Nomad JWKS
- A signing key that is specifically and only for this purpose.
```

What do you think of reformatting to to make this paragraph easier for scanning?

Reply:
I dig the list and rephrasing, but the `KeyID` option is not tied to the scenario list on its own. "The approach" here is "private_key" `KeySource`, which requires one of `KeyID`, `PemCert`, or `PemCertFile`, so "This approach" should be a separate paragraph from `KeyID`. It's a summary of why you might want to consider the entire `#### User provided key` subsection.

Reply:
How's this?

> This approach lets you bring your own RSA key for the following scenarios:
> - Your OIDC provider does not support JWKS
> - Your network topology does not allow connectivity between the provider and Nomad JWKS, even via proxy
> - You want a signing key that is specifically and only for this purpose.